### PR TITLE
T3 overhauling

### PR DESCRIPTION
**Pull request by alongd**
T3 is now written in an OO style instead of functions with global params.
A new input file format was introduced, the executable passes it directly to the API which contains three main sections: `t3`, `rmg`, `qm` (all dictionaries). A pydentic schema was added.
The package was renamed to `t3` instead of `tandem`.

Tests were added to core functionalities, yet moe testing is mandatory before merging.
The documentation must also be updated.

### CODE REVIEW

File: t3/main.py [hunk 1]
```diff
@@ -651,27 +651,26 @@ def determine_species_to_calculate(self) -> bool:
...
+            for species in self.rmg_species:
                 if self.species_requires_refinement(species=species):
                     species_keys.append(self.add_species(species=species, reasons=['All core species']))
         else:
             # SA observables
             for input_species in self.rmg['species']:
                 if input_species['observable'] or input_species['SA_observable']:
                     species_keys.append(self.add_species(
-                        species=get_species_by_label(input_species['label'], rmg_species ),
+                        species=get_species_by_label(input_species['label'], self.rmg_species),
```
Comment (alongd) [inline t3/main.py:666]: Maybe make `get_species_by_label` a method, so it has access to `self.rmg_species`?

**Comment by alongd** [comment]
Yes, I think we instructed it to look for a "tandem" module/folder, we should change it to "t3". I'm not infont of a computer, would you like to try? 
codecov.yml

**Comment by kspieks** [comment]
Yea I'll try changing it now. Do we also need to change `.coveragerc` since that also has `source = tandem`?

**Comment by alongd** [comment]
Nice!!

**Comment by kspieks** [comment]
Of course thanks for the tip! Still getting used to code coverage :)

**Comment by alongd** [comment]
This was a wonderful joint effort.
Looking forward to the next PR!

**Comment by kspieks** [comment]
Same :) I'll work on rebasing the simulate adapters early next week!

**Comment by alongd** [comment]
Great! I'll start working on updating the docs

**Pull request merged**